sql/core/src/main/scala/org/apache/spark/sql/execution/python/BatchEvalPythonExec.scala

```diff
@@ -46,7 +46,7 @@ case class BatchEvalPythonExec(udfs: Seq[PythonUDF], resultAttrs: Seq[Attribute]
     val needConversion = dataTypes.exists(EvaluatePython.needConversionInPython)
 
     // enable memo iff we serialize the row with schema (schema and class should be memorized)
-    val pickle = new Pickler(needConversion)
+    val pickle = new Pickler(needConversion, false)
```

Comment:
Could you leave some comments about why we need `valueCompare=false`?

Reply:
Sure, added in https://github.com/apache/spark/pull/31682/commits/94be320ad32645cc834ae9dd7f1156818c977a10

Reply:
> so we need to disable this feature explicitly (`valueCompare=false`).

I think it is not disabled, but just not compare with value when memorizing?

Reply:
No, cache (by reference) is not disabled. That is the first parameter of 'Pickler'. But if "cache by value" sounds confusing then I can rephrase the comment.

Reply:
I've added a short comment on cache by reference:  https://github.com/apache/spark/pull/31682/commits/bb312316f9de7739997d74d03dddcdb61bf74ec1.

Reply:
Thank you.